File: javatest/omnia/data/cache/CachedIntTest.java
package omnia.data.cache;

import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class CachedIntTest {

  @Test
  public void cache_didReturnSimpleIntCache() {
    assertTrue(CachedInt.cache(() -> 132) instanceof SimpleIntCacher);
  }
}

File: omnia/src/commonTest/omnia/data/cache/CachedIntTest.kt
package omnia.data.cache

import kotlin.test.Test
import omnia.util.test.fluent.Assertion.Companion.assertThat
import omnia.util.test.fluent.isA

class CachedIntTest {

  @Test
  fun cache_didReturnSimpleIntCache() {
    assertThat(CachedInt.cache { 132 }).isA(SimpleIntCacher::class)
  }
}